store/user/dashboard.js

```diff
@@ -15,6 +25,58 @@ class DashboardUser extends User {
       (e) => e.name.toLowerCase().search(searchTerm.toLowerCase()) !== -1
     )
   }
+
+  onDataProviderChange(callback) {
+    this.#onDataProviderChangeCallback.push(callback)
+  }
+
+  @action changeDataProvider(id) {
+    // Clean-up or initial request
+    if (id == null) {
+      this.dataProvider = null
+      return
+    }
+
+    // Compare strings and numbers
+    // since we do not really know what type of ID the API exposes
+    // eslint-disable-next-line eqeqeq
+    const hasTargetId = (dataProvider) => id == dataProvider.id
+
+    // Probably a repeated request. No need to change
+    if (this.dataProvider != null && hasTargetId(this.dataProvider)) return
+
+    // Check access rights
+    const dataProvider = this.dataProviders.find(hasTargetId)
+    if (dataProvider == null) {
+      const dpStr = `DataProvider#${id}`
+      throw new AccessError(`${this.user} does not have access to the ${dpStr}`)
+    }
+
+    this.dataProvider = dataProvider
+    this.#onDataProviderChangeCallback.forEach((callback) => callback())
+  }
+
+  @action
+  async updateDataProvider(patch) {
+    try {
+      const url = `/data-providers/${this.dataProvider.id}`
+      const { data } = await this.options.request(url, {
+        method: 'PATCH',
+        body: patch,
+      })
+      Object.assign(this.user.dataProvider, data)
+    } catch (networkOrAccessError) {
+      // Ignore errors for this moment
+    }
+  }
+
+  @action
+  async updateOrganization(patch) {
+    const { name: institution } = patch
+    this.updateDataProvider({ institution })
+
+    // TODO: Should be a method without cross-call to another method
+  }
```

Comment:
This still should not be a part of the `User` entity.

Reply:
So you would leave it in root?

Reply:
I think it stay as is for the time being. We need a new DataProvider/Organisation substore.

Reply:
I do not understand the reason for moving it here. It was designed as desired before. The data provider is a member of the application model but not of the user. The application controls all changes/switches, updates, loads and unloads stuff. If the user can discover it's own data providers it's exposed to the application and the application decides based on the user's list should the user access the data provider or not.

Reply:
Isn't it kind of breaking single responsibility thing? We were trying to change user property outside of user entity? I don't know and I truly I don't see a big difference in either.

Reply:
We don't. The _User_ has a list of things it can access. The _Application_ is guided (read-only) by this list but still decides what to do.

Furthermore, the _User_ is a part of the _Application_, it should be simple and the _Application_ should be more complex.

I do not think this has anything to the SRP. The SRP claims that the class should be changed because of one and only one actor. I am struggling to define actors here but my guess is that's API for the _BaseUser_ and the Dashboard application for the _DashboardUser_.

Reply:
Ohh I called it wrong I guess. My point was to change things only inside of the object. Anyway so is this something why the PR cannot be merged? Because you are giving me only comment reviews so I have no idea what I am supposed to do.

Reply:
Hope it's now ok with you.